components/brave_shields/browser/tracking_protection_service.cc

```diff
@@ -90,6 +178,78 @@ bool TrackingProtectionService::ShouldStartRequest(const GURL& url,
   return false;
 }
 
+void TrackingProtectionService::DispatchBlockedEvent(int render_process_id,
+  int render_frame_id, const GURL& request_url) {
```

Comment:
Fix: This is incorrect, this displays the metrics on the first-party tracker instead of the `starting_site`.

Reply:
Context: This code dispatches an event to update the tracker count in shields. Since, this event is dispatched as soon as the first party is tagged as a tracker, the count is displayed for the tracker and not the site that initiated the redirect-loop. 

I was thinking it would make sense to dispatch the event only if the `starting_site` is loaded again in the redirect loop instead of dispatching it right away. Thoughts? 

cc: @bridiver @snyderp

Reply:
This is being tracked as a separate issue: https://github.com/brave/brave-browser/issues/3019